plugins/node/instrumentation-kafkajs/src/types.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Span } from '@opentelemetry/api';
+import { InstrumentationConfig } from '@opentelemetry/instrumentation';
+import type { Message } from 'kafkajs';
+
+export interface KafkaProducerCustomAttributeFunction {
+  (span: Span, topic: string, message: Message): void;
+}
+
+export interface KafkaConsumerCustomAttributeFunction {
+  (span: Span, topic: string, message: Message): void;
+}
```

Comment:
@blumamir what do you think about my comment on using generics and having the `Info` as a default?

Reply:
I think it's a good idea and we should apply it in all instrumentations, especially new ones.

Regardless of the type of `Message` object itself, there are also additional info that is captured in this hook: the topic name which is not part of the `message` and perhaps additional data that will be helpful for users in the future.

The motivation for the `Info` object was that we don't end up with hooks with 5-6 arguments that all have to be written even if just one is to be used.
For example, in some instrumentations the hook can also capture the instrumented package versions (e.g. `kafkajs`), like [`amqplib`](https://github.com/open-telemetry/opentelemetry-js-contrib/blob/1af49014b8a77d34058750ce2a87bd58211ebe70/plugins/node/instrumentation-amqplib/src/types.ts#L33). If this would be added later here, and possibly more arguments in the future that users might need, the hook signature can grow over time and become harder to manage as the number of arguments grows. Doing this refactor for existing instrumentations might be a breaking change for users, thus it's a good chance to introduce this pattern here if we think it is appropriate.

Reply:
I've refactored it so the function takes in an info object (with the type defaulting to `KafkajsMessage`)